File: bin_writer.py
import struct
import numpy as np

def generate_random_float_array(size):
    # 랜덤한 32비트 float 배열 생성
    float_array = np.random.rand(size).astype(np.float32).tolist()
    return float_array

def write_float_array_to_file(file_path, float_array):
    # 'f'는 32비트 float를 나타내는 포맷 문자입니다.
    format_string = 'f' * len(float_array)

    # float 배열을 이진 데이터로 패킹합니다.
    packed_data = struct.pack(format_string, *float_array)

    # 이진 데이터를 파일에 쓰기 모드로 엽니다.
    with open(file_path, 'wb') as file:
        # 파일에 패킹된 데이터를 씁니다.
        file.write(packed_data)

# 예제로 사용할 크기 지정
array_size = 100000000 * 3

# 랜덤한 32비트 float 배열 생성
random_float_array = generate_random_float_array(array_size)

# 파일에 쓰기
file_path = 'data_1000M.bin'
write_float_array_to_file(file_path, random_float_array)

print(f'랜덤 Float 배열이 {file_path} 파일에 성공적으로 쓰여졌습니다.')

File: bin_writer.cs
using System;
using System.IO;

class Program
{
    static void Main()
    {
        // 예제로 사용할 크기 지정
        int arraySize = 10;

        // 랜덤한 32비트 float 배열 생성
        float[] randomFloatArray = GenerateRandomFloatArray(arraySize);

        // 파일에 쓰기
        string filePath = "randomFloatArray.bin";
        WriteFloatArrayToFile(filePath, randomFloatArray);

        Console.WriteLine($"랜덤 Float 배열이 {filePath} 파일에 성공적으로 쓰여졌습니다.");
    }

    static float[] GenerateRandomFloatArray(int size)
    {
        Random random = new Random();
        float[] floatArray = new float[size];

        for (int i = 0; i < size; i++)
        {
            // 랜덤한 32비트 float 값 생성
            floatArray[i] = (float)random.NextDouble();
        }

        return floatArray;
    }

    static void WriteFloatArrayToFile(string filePath, float[] floatArray)
    {
        // 파일에 쓰기 모드로 BinaryWriter 열기
        using (BinaryWriter writer = new BinaryWriter(File.Open(filePath, FileMode.Create)))
        {
            // 배열의 각 요소를 파일에 쓰기
            foreach (float value in floatArray)
            {
                writer.Write(value);
            }
        }
    }
}
